helm/opendistro-es/templates/curator/curator.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ template "opendistro-es.fullname" . }}-curator
+  labels:
+    app: curator
+{{ include "opendistro-es.labels.standard" . | indent 4 }}
+spec:
+  schedule: {{ .Values.curator.schedule | quote }}
+  concurrencyPolicy: Forbid
+  startingDeadlineSeconds: {{ .Values.curator.startingDeadlineSeconds }}
+  failedJobsHistoryLimit: {{ .Values.curator.failedJobsHistoryLimit }}
+  successfulJobsHistoryLimit: {{ .Values.curator.successfulJobsHistoryLimit }}
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          restartPolicy: {{ .Values.curator.restartPolicy }} 
+          {{- if .Values.curator.tolerations }}
+          tolerations: {{- toYaml .Values.curator.tolerations | nindent 10 }}
+          {{- end }}
+          {{- if .Values.curator.nodeSelector }}
+          nodeSelector: {{- toYaml .Values.curator.nodeSelector | nindent 12 }}
+          {{- end }}
+          {{- if .Values.curator.affinity }}
+          affinity: {{- toYaml .Values.curator.affinity | nindent 12 }}
+          {{- end }}
+          containers:
+          - name: {{ template "opendistro-es.fullname" . }}-curator
+            image: {{ .Values.curator.image }}
+            imagePullPolicy: {{ .Values.curator.imagePullPolicy }}
+            args:
+            - --config
+            - /etc/config/config.yml
+            - /etc/config/action_file.yml
+            env:
+              - name: ES_USERNAME
+                valueFrom:
+                  secretKeyRef:
+                    name: {{ .Values.kibana.elasticsearchAccount.secret }}
+                    key: username
+              - name: ES_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: {{ .Values.kibana.elasticsearchAccount.secret }}
+                    key: password
+              - name: ES_AUTH
+                value: $(ES_USERNAME):$(ES_PASSWORD)
```

Comment:
I'm not familiar with this, how does it work? Will the value be visible in `kubectl get pods -o yaml`?

Reply:
Checking right now...

Reply:
Nope it does not look like it :slightly_smiling_face: 

From `describe`
```
    Environment:
      ES_USERNAME:  <set to the key 'username' in secret 'opendistro-es-es-account'>  Optional: false
      ES_PASSWORD:  <set to the key 'password' in secret 'opendistro-es-es-account'>  Optional: false
      ES_AUTH:      $(ES_USERNAME):$(ES_PASSWORD)
```

From `get`
```
❯ kubectl get pods opendistro-es-curator-1589979600-tk6bx -ojson | jq -r '.spec.containers[].env'
[
  {
    "name": "ES_USERNAME",
    "valueFrom": {
      "secretKeyRef": {
        "key": "username",
        "name": "opendistro-es-es-account"
      }
    }
  },
  {
    "name": "ES_PASSWORD",
    "valueFrom": {
      "secretKeyRef": {
        "key": "password",
        "name": "opendistro-es-es-account"
      }
    }
  },
  {
    "name": "ES_AUTH",
    "value": "$(ES_USERNAME):$(ES_PASSWORD)"
  }

```

Reply:
I can add that the reason for the `ES_AUTH` variable is because `${USERNAME}:${PASSWORD}` does not get interpolated correctly by curator :slightly_frowning_face: .

Reply:
Good then I'm fine with this :+1: